content/articles/js-router/index.md

```diff
@@ -0,0 +1,170 @@
+### How to create a simple router in JavaScript using window event listeners
+A router is a JavaScript object that maps URLs to functions. The router calls a function based on the URL. It is also used for creating the URL based on a function.
+
+Before the rise of single page applications, a web application was a series of interconnected pages. This could be either static pages or pages dynamically generated in the server.
+
+Single page applications was a new concept where a web application's life span evolves around a single html document. Transition between pages was replaced by use of different views. Links are no longer used to generate and navigate between pages, instead they are bound to views.
+
+A view is a JavaScript object that contains the logic for a single page. The view handles page rendering and user input. All the views are controlled by the router.
+
+### The concept
+A router watches for changes in the URL and calls the appropriate function. The function is responsible for rendering the view.
+
+Consider a router as one big function that handles all the routing logic. The router is responsible for calling some pre-defined function/s mapped to the given route. 
+
+For example, if the URL is `/about`, the router calls the `about` function/s. The called function will call other functions that deal with getting data from backend services or DOM manipulation. They tear down the previous view and render the new view.
+
+A view can be as simple as a function that creates a `div` element and appends it to the body.
+
+To demonstrate this concept, let's create a simple router in JavaScript.
+
+Create two files:
+1. `index.html`
+2. `script.js`
+
+In your `index.html`, create a `div` with the id: "app". Also link the `script.js` file.
+
+```html
+<div id="app"></div>
+<script src="./script.js"></script>
+```
+
+That's all for the `index.html` file.
+
+In your `script.js` file, create two objects to hold the routes and the template functions to be rendered.
+
+```javascript
+let routes = {};
+let templates = {};
+```
+
+Then create two functions to display the home page and the about page. You can use arrow functions to define these functions, but I'll use regular functions for simplicity.
+
+```javascript
+let app_div = document.getElementById('app');
+
+function home() {
+    let div = document.createElement('div');
+    let link = document.createElement('a');
+    link.href = '#about';
+    link.innerText = 'About';
+
+    div.innerHTML = '<h1>Home</h1>';
+    div.appendChild(link);
+
+    app_div.appendChild(div);
+};
+
+function about() {
+    let div = document.createElement('div');
+    let link = document.createElement('a');
+    link.href = '#home';
+    link.innerText = 'Home';
```

Comment:
The link `#home` should be `#/`. There is no route defined as `home` in your code.